pkg/handlers/app.go

```diff
@@ -341,11 +341,21 @@ func responseAppFromApp(a *apptypes.App) (*types.ResponseApp, error) {
 				return nil, errors.Wrap(err, "failed to check if cluster requires upgrade")
 			}
 
-			embeddedClusterInstallation, err := embeddedcluster.GetCurrentInstallation(context.TODO())
+			embeddedClusterInstallations, err := embeddedcluster.ListInstallations(context.TODO())
 			if err != nil {
-				return nil, errors.Wrap(err, "failed to get current installation")
+				return nil, errors.Wrap(err, "failed to list installations")
+			}
+
+			cluster.NumInstallations = len(embeddedClusterInstallations)
+
+			latestEmbeddedClusterInstallation, err := embeddedcluster.GetLatestInstallation(context.TODO(), embeddedClusterInstallations)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get latest installation")
+			}
+
+			if latestEmbeddedClusterInstallation != nil {
+				cluster.State = string(latestEmbeddedClusterInstallation.Status.State)
```

Comment:
if the installation was just recently created, `.Status.State` may be empty. the goal here is to provide an additional piece of data `numInstallations` to the UI so it can make the proper determination of whether or not to show the modal in these cases if downtime occurs before it receives an actual state.